hearbaseline/vqt.py

```diff
@@ -0,0 +1,140 @@
+"""
+variable-Q transform spectrogram (60 bins/octave, ~10ms step)
+
+Based upon DCASE 2016 Task 2 baseline.
+The scene embeddings will be an average, i.e. only 84 dimensions.
+"""
+
+import math
+import numpy as np
+from typing import Tuple
+
+import librosa
+import torch
+from torch import Tensor
+
+
+class VQT(torch.nn.Module):
+    # Not really a torch module
+
+    # sample rate and embedding sizes are required model attributes for the HEAR API
+    sample_rate = 44100
+    # dcase baseline was 10 ms (441), but for librosa hop_length must be
+    # a positive integer multiple of 2^8 for 9-octave CQT/VQT
+    hop_length = 512
+    gamma = 30
+    bins_per_octave = 60
+    fmin = 27.5
+    noctaves = int(math.log2(sample_rate / 2 / fmin))
+
+    embedding_size = bins_per_octave * noctaves
```

Comment:
Could leave the noctaves as a float and then do the int cast here as well if you wanted to get closer to fs/2 (closer to dcase I believe)-- but not sure if that is necessary. 9 octaves goes up to 14kHz so that is probably plenty?

Reply:
librosa makes this requirement I believe